pkgs/tools/inputmethods/skk/skk-dicts/default.nix

```diff
@@ -1,77 +1,202 @@
-    sha256 = "1vrwnq0vvjn61nijbln6wfinqg93802d2a8d4ad82n692v83b1li";
-  };
-  # misc
-  assoc = fetchurl {
-    url = "https://raw.githubusercontent.com/skk-dev/dict/8b35d07a7d2044d48b063d2774d9f9d00bb7cb48/SKK-JISYO.assoc";
-    sha256 = "1smcbyv6srrhnpl7ic9nqds9nz3g2dgqngmhzkrdlwmvcpvakp1v";
-  };
-in
+  mkDictNameValue =
+    { name
+    , description
+    , license # it's written in the beginning of each file
+    , files ? [ "SKK-JISYO.${name}" ]
+    }: {
+      name = lib.toLower (builtins.replaceStrings [ "." ] [ "_" ] name);
+      value = stdenvNoCC.mkDerivation {
+        pname = "skk-jisyo-" + lib.toLower name;
+        version = "0-unstable-2024-01-31";
 
-stdenv.mkDerivation {
-  pname = "skk-dicts-unstable";
-  version = "2020-03-24";
-  srcs = [ small medium large edict assoc ];
-  nativeBuildInputs = [ iconv skktools ];
+        src = fetchFromGitHub {
+          owner = "skk-dev";
+          repo = "dict";
+          rev = "b798a46b886f71c0c25ad2a9e78b1c3e8933970c";
+          sha256 = "sha256-IvQFi/VA+klwolJGRyxAgT/ZOI5EYRNvXfAqWGhTTKM=";
+        };
 
-  strictDeps = true;
+        nativeBuildInputs = lib.optionals useUtf8 [ nkf ];
 
-  dontUnpack = true;
+        strictDeps = true;
 
-  installPhase = ''
-    function dictname() {
-      src=$1
-      name=$(basename $src)          # remove dir name
-      dict=$(echo $name | cut -b34-) # remove sha256 prefix
-      echo $dict
-    }
-    mkdir -p $out/share
+        buildPhase = ''
+          runHook preBuild
+        '' + lib.concatMapStrings
+          (file: ''
+            nkf -w ${file} \
+              | LC_ALL=C sed 's/coding: [^ ]\{1,\}/coding: utf-8/' \
+              > ${file + suffix}
+          '')
+          (lib.optionals useUtf8 (map lib.escapeShellArg files)) + ''
+          runHook postBuild
+        '';
 
-    for src in $srcs; do
-      dst=$out/share/$(dictname $src)
-      echo ";;; -*- coding: utf-8 -*-" > $dst  # libskk requires this on the first line
-      iconv -f EUC-JP -t UTF-8 $src | skkdic-expr2 >> $dst
-    done
+        installPhase = ''
+          runHook preInstall
+        '' + lib.concatMapStrings
+          (file: ''
+            install -Dm644 \
+              ${lib.escapeShellArg file} \
+              $out/share/skk/${lib.escapeShellArg (baseNameOf file)}
+          '')
+          (map (file: file + suffix) files) + ''
+          runHook postInstall
+        '';
 
-    # combine .L .edict and .assoc for convenience
-    dst=$out/share/SKK-JISYO.combined
-    echo ";;; -*- coding: utf-8 -*-" > $dst
-    skkdic-expr2 \
-      $out/share/$(dictname ${large}) + \
-      $out/share/$(dictname ${edict}) + \
-      $out/share/$(dictname ${assoc}) >> $dst
-  '';
+        doInstallCheck = true;
+        installCheckPhase = ''
+          emptydict=': 0 candidates$'
+          ${skktools}/bin/skkdic-count /dev/null | grep "$emptydict"
+          ${skktools}/bin/skkdic-count $out/share/skk/* | grep -v "$emptydict"
+        '';
```

Comment:
I added a new test to see if there are any actual entries in the dictionary files.